src/Build/BackEnd/Components/RequestBuilder/TaskHost.cs

```diff
@@ -676,6 +676,38 @@ public void LogTelemetry(string eventName, IDictionary<string, string> propertie
         public bool AllowFailureWithoutError { get; set; } = false;
         #endregion
 
+        #region IBuildEngine8 Members
+        private ICollection<string> _warningsAsErrors;
+
+        /// <summary>
+        /// Contains all warnings that should be logged as errors.
+        /// Non-null empty set when all warnings should be treated as errors.
+        /// </summary>
+        private ICollection<string> WarningsAsErrors
+        {
+            get
+            {
+                // Test compatibility
+                if(_taskLoggingContext == null)
+                {
+                    return null;
+                }
+
+                return _warningsAsErrors ??= _taskLoggingContext.GetWarningsAsErrors();
+            }
+        }
+
+        public bool ShouldTreatWarningAsError(string warningCode)
+        {
+            if (_taskLoggingContext == null || WarningsAsErrors == null)
```

Comment:
nit:
```suggestion
            if (WarningsAsErrors == null)
```

(Should be the same but looks a little cleaner to me.)